src/test/spec/v2_runner/test_file.rs

```diff
@@ -27,10 +34,12 @@ pub struct TestFile {
 
 #[derive(Debug, Deserialize)]
 #[serde(rename_all = "camelCase")]
+#[serde(deny_unknown_fields)]
 pub struct RunOn {
     pub min_server_version: Option<String>,
     pub max_server_version: Option<String>,
     pub topology: Option<Vec<String>>,
+    pub(crate) serverless: Option<Serverless>,
```

Comment:
As far as I can tell, this is all unexported test code, so `pub` and `pub(crate)` are functionally equivalent; `pub(crate)` is needed here because the `Serverless` type is itself that, which in turn is because that's what the local file it's in uses for other things.  Might be worth doing a pass to make everything in src/test one or the other.

Reply:
I personally try to only use `pub` for things that are truly public, even in test / private modules, though I think we didn't always do that.

Reply:
I don't have a strict policy on how I handle these things but it would be nice to have a convention -- I think `pub(crate)` on everything that's not actually user-facing is the way to go.